### ISSUE TITLE
Getting the data takes more time than calculating it

### ISSUE THREAD

**Issue by Littlesharkcug** [RL-VIG/LibFewShot#29]
![image](https://user-images.githubusercontent.com/40341191/142340212-2c92990d-011b-4f59-9cae-84391b95a601.png)

**Comment by wZuck** [comment]
It's hard to say what's going wrong. Could you please paste your config so we can try to reproduce it?

**Comment by wZuck** [comment]
It is caused by not enough num_workers in the dataloader (especially [episodic-dataloader](https://github.com/RL-VIG/LibFewShot/blob/97d7100687d526a853604d882a7d0339ddd13541/core/data/dataloader.py#L101)), you can modify `num_workers=config["n_gpu"] * 4` to other values, we suggest you use `*8` for some methods with `Conv64F` backbone.

**Comment by Littlesharkcug** [comment]
![image](https://user-images.githubusercontent.com/40341191/142877659-e8fcbee4-a732-4204-9810-ef026df9d635.png)
![image](https://user-images.githubusercontent.com/40341191/142878129-f499b779-ed36-48f0-9969-7f9de904b0cf.png)

augment: true
augment_times: 1
augment_times_query: 1
backbone:
  kwargs:
    is_feature: false
    is_flatten: false
    last_pool: false
    leaky_relu: false
    maxpool_last2: false
    negative_slope: 0.2
  name: Conv64F
batch_size: 128
classifier:
  kwargs:
    feat_dim: 64
    feat_height: 3
    feat_width: 3
  name: RelationNet
data_root: data/miniImageNet--ravi
deterministic: true
device_ids: 0
episode_size: 1
epoch: 50
image_size: 84
includes:
- headers/data.yaml
- headers/device.yaml
- headers/misc.yaml
- headers/model.yaml
- headers/optimizer.yaml
- classifiers/RelationNet.yaml
- backbones/Conv64F.yaml
log_interval: 100
log_level: info
log_name: null
log_paramerter: false
lr_scheduler:
  kwargs:
    T_max: 100
    eta_min: 0
  name: CosineAnnealingLR
n_gpu: 1
optimizer:
  kwargs:
    lr: 0.001
  name: SGD
  other:
    emb_func: 0.001
parallel_part:
- emb_func
pretrain_path: null
query_num: 15
result_root: ./results
resume: false
save_interval: 10
save_part:
- emb_func
seed: 0
shot_num: 1
tag: null
tb_scale: 1.0
test_episode: 100
test_epoch: 5
test_query: 15
test_shot: 1
test_way: 5
train_episode: 100
use_memory: false
way_num: 5


I have changed num_workers=config["n_gpu"] * 8,
My Plantform is 
![image](https://user-images.githubusercontent.com/40341191/142878269-cfdd209f-530e-46a8-a161-3a07feb17823.png)

**Comment by wZuck** [comment]
Can you checkout to `add_ddp` branch and set `num_workers` to `32`(or the value suitable for your platform)? we have tested it and found it fast enough.

**Comment by Littlesharkcug** [comment]
> Can you checkout to `add_ddp` branch and set `num_workers` to `32`(or the value suitable for your platform)? we have tested it and found it fast enough.

Thank you.  I checkout to `add_ddp` branch but it takes  more time. And I checkout to `main`branch and  set num_workers to 32, the program runs faster.
